### DISCUSSION TITLE
sphinx-autobuild Infinite Loop - any Solution?

### DISCUSSION THREAD

**Discussion by michaelweinold** [sphinx-doc/sphinx#12381] [General]
I have been using `sphinx-autobuild` with all our documentation projects. Unfortunately, I can not figure out how to avoid the infinite build loops, which seem to be affecting most users:

Related issues currently:

 - https://github.com/sphinx-doc/sphinx-autobuild/issues/154
 - https://github.com/sphinx-doc/sphinx-autobuild/issues/148
 - https://github.com/sphinx-doc/sphinx-autobuild/issues/120
 - https://github.com/sphinx-doc/sphinx-autobuild/issues/131
 - https://github.com/sphinx-doc/sphinx-autobuild/issues/117
 - https://github.com/sphinx-doc/sphinx-autobuild/issues/112

In our documentation, I am using `sphinx-autoapi`. The package creates an `api` folder in our directory structure:

```
.
└── source/
    ├── conf.py
    ├── index.md
    └── content/
        ├── introduction/
        │   ├── chapter_1.md
        │   └── chapter_2.md
        └── api/
            ├── autogenerated_api_page_1.html
            └── autogenerated_api_page_2.html
```

But commands like

```bash
sphinx-autobuild --ignore=source/content/api source _build/html -j auto --open-browser
```

don't work. The `sphinx-autobuild` package just keeps on building the documentation in an infinite loop.

There are two PRs currently open, which might solve all these issues:
 - https://github.com/sphinx-doc/sphinx-autobuild/pull/135
 - https://github.com/sphinx-doc/sphinx-autobuild/pull/123

Related issues:
 - https://github.com/sphinx-doc/sphinx-autobuild/issues/140
 - https://github.com/sphinx-doc/sphinx-autobuild/issues/120

Could a package maintainer take a look? Any help would be greatly appreciated!

**Comment by tngTUDOR** [comment]
@michaelweinold , on eof the 2 PRs you mentionned has been superseeded by another one:

123 got supersseded by: https://github.com/sphinx-doc/sphinx-autobuild/pull/171 (not merged yet tho :( )

**Comment by cmarqu** [comment]
Maybe https://github.com/sphinx-doc/sphinx-autobuild/issues/188 gives you a workaround.
